cartoframes/data/services/bq_datasets.py

```diff
@@ -43,6 +112,48 @@ def download_stream(self):
         return ResponseStream(self.download())
 
 
+class BQJob:
+
+    def __init__(self, job_id, name_id):
+        self.id = job_id
+        self.name = name_id
+        # TODO fix this crap
+        self.session = requests.Session()
+        self.api_key = 'my_valid_api_key'
+
+
+    def status(self):
+        url = DO_ENRICHMENT_API_URL + '/datasets/' + self.name + '/imports/' + self.id
+        params = {'api_key': self.api_key}
+
+        try:
+            response = self.session.get(url, params=params)
+            response.raise_for_status()
+
+            body = response.json()
+
+            return body['status']
+        except requests.HTTPError as e:
+            if 400 <= response.status_code < 500:
+                # Client error, provide better reason
+                reason = response.json()['error'][0]
+                error_msg = u'%s Client Error: %s' % (response.status_code,
+                                                      reason)
+                raise CartoException(error_msg)
+            else:
+                raise CartoException(e)
+        except Exception as e:
+            raise CartoException(e)
+
+    def result(self):
+        status = self.status()
+
+        while status != 'done':
```

Comment:
is `done` the only possible terminal status?

Reply:
There is no method to know if the job has failed. https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.LoadJob.html#google.cloud.bigquery.job.LoadJob

I guess I can do something  with `.errors` attribute. But It will require some changes in DO API. Giving a try.

Reply:
Done https://github.com/CartoDB/cartoframes/pull/1513/commits/d16889b7ebfa383d2ef643b1df2c4cd372347ec3